Base-Commit: fc65416c22c47e15be1daf2e2d62deaba6881591
Base-Branch: 
diff --git a/download.go b/download.go
index 16b6108..6bdbd32 100644
--- a/download.go
+++ b/download.go
@@ -415,3 +415,20 @@ func validateMedia(iv, file, macKey, mac []byte) error {
 	}
 	return nil
 }
+
+func (cli *Client) DownloadMediaWithUrl(ctx context.Context, url string, mediaKey []byte, appInfo MediaType, fileLength int, fileEncSha256 []byte, fileSha256 []byte) (data []byte, err error) {
+	return cli.downloadAndDecrypt(ctx, url, mediaKey, appInfo, fileLength, fileEncSha256, fileSha256)
+}
+
+func GetDownloadSize(msg DownloadableMessage) int {
+	return getSize(msg)
+}
+
+func GetMMSType(mediaType MediaType) string {
+	return mediaTypeToMMSType[mediaType]
+}
+
+type DownloadableMessageWithURL interface {
+	DownloadableMessage
+	GetUrl() string
+}
